daemon/cmd/daemon_main.go

```diff
@@ -360,6 +360,15 @@ func initializeFlags() {
 	flags.String(option.NodeEncryptionOptOutLabels, defaults.NodeEncryptionOptOutLabels, "Label selector for nodes which will opt-out of node-to-node encryption")
 	option.BindEnv(Vp, option.NodeEncryptionOptOutLabels)
 
+	flags.Bool(option.EnableEncryptionStrictMode, false, "Enable encryption strict mode")
+	option.BindEnv(Vp, option.EnableEncryptionStrictMode)
+
+	flags.String(option.EncryptionStrictModeCIDR, "", "In strict-mode encryption, all unencrypted traffic coming from this CIDR and going to this same CIDR will be dropped")
+	option.BindEnv(Vp, option.EncryptionStrictModeCIDR)
+
+	flags.Bool(option.EncryptionStrictModeAllowDynamicLookup, false, "EncryptionStrictModeAllowDynamicLookup allows dynamic lookup of node identities. This is required when tunneling is used or direct routing is used and the node CIDR and pod CIDR overlap.")
```

Comment:
This is a bit of bikeshedding, but I wonder if there is a better name for this flag. The documentation on this good on when it's needed, but just from the name I was a bit confused on what it does and why I would use it.

The main issue here is that as a user, it's not clear to me what it means to allow a dynamic lookup. What are the consequences of allowing a lookup? What will the lookup do?

I would propose the following name instead:

- `strict-mode-allow-node-identities: Allows unencrypted traffic from pods to remote node identities within the strict mode CIDR`" This makes it very explicit what the code currently does if the flag is enabled to true.

There are alternatives of course, we could also name it `strict-mode-allow-known-identities` if you we assume that we will have to add other exceptions in the future (e.g. toCIDR targets within the strict CIDR maybe?). Or `strict-mode-allow-remote-node-identities` if we want to be really precise.

Reply:
Thanks for the detailed train of thought! I was also unhappy with the current name, but I really like your proposals. Since I hope that we don't need to soften the strict mode even more in the future, I'll go with `strict-mode-allow-remote-node-identities` .